### DISCUSSION TITLE
Acceptable types is very strict 0.23.2

### DISCUSSION THREAD

**Discussion by j-carson** [encode/httpx#2538] [Potential Issue] [answered]
The change to accepted argument types in version 0.23.2 issue #2523 broke my code. 

In my case, it's a datetime.date that isn't being cast to a string, and date strings are allowed in OpenAPI so this list of accepted types (str, int, float, bool, None)  seems too strict to me. 

This change will also break things that are int-like or float-like (say numpy types like np.int32) rather than pure-python numbers, even though it's perfectly safe to case these to strings.

**Comment by zanieb** [comment, marked as answer]
Thanks for the note! I think more discussion here is merited.

I think this is further encouragement for reverting the change and including it in a future release. See https://github.com/encode/httpx/pull/2523#issuecomment-1370036069 and https://github.com/encode/httpx/pull/2539

**Reply by lovelydinosaur** [reply]
Thanks folks.

@j-carson We've now reverted this and released 0.23.3 - https://pypi.org/project/httpx/0.23.3/

**Reply by j-carson** [reply]
Thank you!
